### PR TITLE
Maven central release setup

### PR DESCRIPTION
**Pull request by JockX**
Updates the project to conform with Maven Central requirements

**Edit:** See also https://github.com/java-native/jssc/issues/23#issuecomment-665620422

### CODE REVIEW

**Comment by JockX** [comment]
This PR builds upon #66, and I'd rebase this one if the other is accepted. Opening for the purpose of review.
Addresses some points mentioned in #23

File: pom.xml [hunk 1]
```diff
@@ -4,13 +4,40 @@
   xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
   <modelVersion>4.0.0</modelVersion>
 
-  <groupId>io.github.java-native.jssc</groupId>
+  <groupId>io.github.java-native</groupId>
```
Comment (JockX) [inline pom.xml:7]: As per https://issues.sonatype.org/browse/OSSRH-54753

File: pom.xml [hunk 2]
```diff
@@ -42,25 +69,11 @@
...
-    <plugin.javadoc.version>3.0.1</plugin.javadoc.version>
+    <plugin.javadoc.version>3.1.1</plugin.javadoc.version>
     <plugin.source.version>3.0.1</plugin.source.version>
     <plugin.compiler.version>3.8.0</plugin.compiler.version>
   </properties>
 
-  <organization>
-    <name>Java Native</name>
-    <url>https://github.com/java-native/</url>
-  </organization>
```
Comment (JockX) [inline pom.xml:54]: I allowed myself to remove the organization tag, as it is not mandatory, and does not really represents an organization.

File: pom.xml [hunk 3]
```diff
@@ -42,25 +69,11 @@
...
-
-  <licenses>
-    <license>
-      <name>GNU LGPL 3</name>
-      <url>http://www.gnu.org/licenses/lgpl.txt</url>
-      <distribution>repo</distribution>
-    </license>
-  </licenses>
```
Comment (JockX) [inline pom.xml:62]: Moved up to conform with standard pom.xml layout

File: pom.xml [hunk 4]
```diff
@@ -4,13 +4,40 @@
   xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
   <modelVersion>4.0.0</modelVersion>
 
-  <groupId>io.github.java-native.jssc</groupId>
+  <groupId>io.github.java-native</groupId>
   <artifactId>jssc</artifactId>
-  <version>2.9.1</version>
+  <version>2.9.2-SNAPSHOT</version>
```
Comment (JockX) [inline pom.xml:9]: As the version 2.9.1 is considered released, but not up to Maven Central standards, I'd suggest bumping it to 2.9.2, and keeping the SNAPSHOT suffix between releasable revisions..

File: pom.xml [hunk 5]
```diff
@@ -42,25 +69,11 @@
     <plugin.osmaven.version>1.6.2</plugin.osmaven.version>
     <plugin.signature.version>1.1</plugin.signature.version>
     <plugin.surfire.version>3.0.0-M3</plugin.surfire.version>
-    <!-- keep below 3.1.0 because 3.1.0 fails on java 8. -->
-    <plugin.javadoc.version>3.0.1</plugin.javadoc.version>
+    <plugin.javadoc.version>3.1.1</plugin.javadoc.version>
```
Comment (tresf) [inline pom.xml:72]: Can you please explain?

Comment (tresf) [reply pom.xml:72]: I see...

> This PR builds upon #66

I'll ask questions there.

**Comment by tresf** [comment]
My main concern with this PR is that the native libraries will still be placed in the right location.  It'd also be nice to see Travis-CI in a better state (to no fault of yours) before merging.

For starters, Travis bumped the build environment (14.04 -> 16.04), so `cmake3` is now simply `cmake`.  macOS seems to be failing with the following, which I haven't had time to research.

> ```
> [ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.8.0:compile (default-compile) on project jssc: Fatal error compiling: error: release version 6 not supported -> [Help 1]
> ```

**Comment by JockX** [comment]
Looks like Travis is building with JDK12+ while the compilation target of 1.6 is only supported up to JDK11(LTS). 
FYI, This PR was tested with Corretto JDK8 and Corretto JDK11, but got the same error on 12.

> [ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.8.0:compile (default-compile) on project jssc: Fatal error compiling: error: release version 6 not supported -> [Help 1]

**Comment by JockX** [comment]
> My main concern with this PR is whether or not the native libraries will still be placed in the right location.
`mvn clean install` on root directory causes jssc.dll to be updated under `src/main/resources-precompiled/natives/windows_64` on my Windows 10 using Visual Studio's CMAKE, so I assume nothing is broken (although more detailed setup instructions for windows would be welcome :) ).  Ideally it should be created under target, but this is outside of scope for change.

**Comment by tresf** [comment]
> Looks like Travis is building with JDK12+ while the compilation target of 1.6 is only supported up to JDK11(LTS).
FYI, This PR was tested with Corretto JDK8 and Corretto JDK11, but got the same error on 12.

Fixed via d0c6223.

**Comment by tresf** [comment]
> Ideally it should be created under target, but this is outside of scope for change.

This was done intentionally via #55 although I'm open to ideas.

> although more detailed setup instructions for windows would be welcome 

Between Travis, Maven and CMake, the build environment is quite hard for even me to follow, especially when it's only touched on occasion.  From what I recall correctly, the system was only written with developers (not releases) in mind and assumed a standard Maven setup, CMake installed from the website and some -- any version -- of MSVC available.  I'm sure this can be improved and as I begin working on `arm64/aarch64`, I'll be setting this environment up again.  Any specific questions are welcome.

**Pull request merged**